utils/yolo/converter.py

```diff
@@ -0,0 +1,264 @@
+            auto_lbl_count+=1
+
+    tracks= cvat_xml.findall( './/track' )
+
+    if (tracks is not None) and (len(tracks) > 0):
+        frames = {}
+
+        for track in tracks:
+            trackid = int(track.get("id"))
+            label = track.get("label")
+            boxes = track.findall( './box' )
+            for box in boxes:
+                frameid  = int(box.get('frame'))
+                outside  = int(box.get('outside'))
+                #occluded = int(box.get('occluded'))  #currently unused
+                #keyframe = int(box.get('keyframe'))  #currently unused
+                xtl      = float(box.get('xtl'))
+                ytl      = float(box.get('ytl'))
+                xbr      = float(box.get('xbr'))
+                ybr      = float(box.get('ybr'))
+
+                frame = frames.get( frameid, {} )
+
+                if outside == 0:
+                    frame[ trackid ] = { 'xtl': xtl, 'ytl': ytl, 'xbr': xbr, 'ybr': ybr, 'label': label }
+
+                frames[ frameid ] = frame
+
+        width = int(cvat_xml.find('.//original_size/width').text)
+        height  = int(cvat_xml.find('.//original_size/height').text)
+
+        taskid = int(cvat_xml.find('.//task/id').text)
+
+        urlsegment = cvat_xml.find(".//segments/segment/url").text
+        urlbase = urlsegment.split("?")[0]
+
+        httpclient = requests.session()
+        httpclient.get(urlbase)
+
+        csrftoken = "none"
+        sessionid = "none"
+
+        # Spit out a list of each object for each frame
+        for frameid in sorted(frames.keys()):
+            image_name = "%s_%08d.jpg" % (basename, frameid)
+            image_path = os.path.join(image_dir, image_name)
+            if not os.path.exists(image_path):
+                if username is None:
+                    log.warn('{} image cannot be found. Is `{}` image directory correct?\n'.format(image_path, image_dir))
+                else:
+                    log.info('{} image cannot be found. Downloading from task ID {}\n'.format(image_path, taskid))
+
+                    if sessionid == "none":
+                        if "csrftoken" in httpclient.cookies:
+                            csrftoken = httpclient.cookies["csrftoken"]
+                        elif "csrf" in httpclient.cookies:
+                            csrftoken = httpclient.cookies["csrf"]
+
+                        login_data = dict(username=username, password=password,
+                                        csrfmiddlewaretoken=csrftoken, next='/dashboard')
+
+                        urllogin = urlbase+"/auth/login"
+                        httpclient.post(urllogin, data=login_data,
+                                        headers=dict(Referer=urllogin))
+
+                        if ("sessionid" in httpclient.cookies):
+                            sessionid = httpclient.cookies["sessionid"]
+
+                    url = urlbase+"/api/v1/tasks/"+str(taskid)+"/frames/"+ str(frameid)
```

Comment:
@dntxos , I general I don't think that it is a good idea to contact CVAT instance to get frames. Frames which CVAT provides using the API don't have good quality. It isn't something which should be used for training. Quality of frames should be enough to do annotation. They can be compressed and this is why we have "image quality" parameter when an annotation task is created.

Reply:
You can think that user provides you all necessary data. Thus your code will be simpler.

Reply:
Yep. This code download frames from CVAT only when a username is passed. It is an optional parameter. Since in my scenario everything works locally and i loaded with good quality (not standard 50), this seemed useful to me. I can remove from the code if you prefer. ;-)

Reply:
@dntxos, if the code is useful for you let's keep it. I will try to test your code and if I don't see any other issues I will merge. Thanks again for your contribution.

Reply:
@nmanovic, cool!  Thx for your feedback.  It is an honor to contribute to this amazing project.  Our annotation time has considerably reduced with CVAT. 🤩👏👏👏